### ISSUE TITLE
Kernel panic with rtai_math if _sincos is not found. 

### ISSUE THREAD

**Issue by andypugh** [LinuxCNC/linuxcnc#893]
If configure finds sincos but not __sincos (?) then HAVE_SINCOS is defined and HAVE__SINCOS is undefined. 

This causes a 100% repeatable kernel panic running the on_abort_command_crazy_move test. But it requires a setup that probably only I  have. I am running Buster with an RTAI "vulcano" patched 4.19.114 kernel with RTAI configured to use the MUSL maths library 

However I think that the logic can be followed by anyone who understands the configure scripts and rtapi_math (ie, not me) 

configure output: 

```
checking for sincos function... yes
checking for __sincos... no
```

dmesg (with debugging turned up and CONFIG_DYNAMIC_DEBUGGING=y)

```
Jun 28 00:46:40 rm-one kernel: [12329.424327] 	0xffffffffc0b8f000 .note.gnu.build-id
Jun 28 00:46:40 rm-one kernel: [12329.424328] 	0xffffffffc0b8f024 .note.Linux
Jun 28 00:46:40 rm-one kernel: [12329.424342] 	0xffffffffc0b74000 .text
Jun 28 00:46:40 rm-one kernel: [12329.424343] 	0xffffffffc0b8f040 .rodata
Jun 28 00:46:40 rm-one kernel: [12329.424344] 	0xffffffffc0b8f0d8 .rodata.cst8
...
Jun 28 00:46:40 rm-one kernel: [12329.424405] motmod does not use rtai_math!
...
Jun 28 00:46:45 rm-one kernel: [12334.686628] RTAPI: Task 1[ffffa42014c86000]: Fault with vec=8, signo=11 ip=ffffffffc0b8e434.
Jun 28 00:46:45 rm-one kernel: [12334.686628] RTAPI: This fault may not be recoverable without rebooting.

```

(Followed by total system lockup) 

addr2line points at sincos.c line 36:

```
#include "config.h"

#if defined(__KERNEL__)
#undef HAVE_SINCOS
#endif

#ifndef HAVE_SINCOS

#include "rtapi_math.h"
#include "sincos.h"

#ifndef HAVE___SINCOS

#include "posemath.h"

void sincos(double x, double *sx, double *cx)
{
    *sx = sin(x);
    *cx = cos(x);
}

#endif

#endif
```

And, indeed, commenting out the calls to sin and cos means that the system does not crash and the on_abort_crazy_move test passes. 

My theory is that this code isn't getting a valid sin and cos. The code definitely runs (proven by adding a printk with the sin and cos commented)

**Comment by andypugh** [comment]
Jepler pointed out that the crash is due to sincos being optimised by the compiler to a call to sincos.

**Comment by andypugh** [comment]
Closed by #896
